### ISSUE TITLE
StateModifier has no effect

### ISSUE THREAD

**Issue by rjmoggach** [gadicc/fview-sizeconstraint#1]
Probably related to the ordering issues you're already seeing - 
Tried to use a StateModifier above a SizeConstraint and it does nothing. Thought a ContainerSurface wrapping the SizeConstraint might work but then the transforms are crazed.

[mogfam.meteor.com](http://mogfam.meteor.com)

[https://github.com/robmoggach/famous-views-testing](https://github.com/robmoggach/famous-views-testing)

-will also post an issue to fview-svg as I was unable to make svgs responsive using that surface wrapper - needs additional css love

**Comment by rjmoggach** [comment]
I also have noooo idea where the skew is coming from. again probably what you're seeing... some broken matrix math i imagine. seen this before in my VFX day job.

Question: PEM commented on my code insinuating very briefly that I shouldn't use ContainerSurface - what's the reason?

**Comment by gadicc** [comment]
Hey, I couldn't find a SizeContraint in the code anywhere... which test/template were you using it on?  What arguments were you giving StateModifier?

I've seem a number of bugs in famo.us until now but nothing involving problems with the transform matrix.  I couldn't find the skew you mentioned (maybe you've removed the code since then) -- I'm curious if it's a SizeConstraint thing and if the same happens in vanilla famo.us outside of Meteor.  But I'd still love an example to look at inside of Meteor.

I'm not sure what in your code PEM was commenting on.  In general you should avoid ContainerSurface if you don't need it, since each one creates a new context and uses more overall CPU.  Basically, you want a ContainerSurface for anywhere you need `overflow: hidden` (which unfortunatelly is actually a bit buggy in some browsers) or a different perspective to elsewhere in your app.  Otherwise if you can size and order your renderables in a way to avoid it, you'll get better performance.

Also, re the ordering issue, it was this:
http://stackoverflow.com/questions/23583896/surface-order-is-messed-when-switching-back-with-famo-us-rendercontroller

In short, we shouldn't rely on DOM ordering in Famo.us (well, unless we know we can... but in a RenderController for example, we can't, usually :)) and should explicitly translate along the z-axis.  Will fix up the demos now.

**Comment by rjmoggach** [comment]
I think the issue was more with getting svgs to play nice in a responsive
capacity - they need a container div to work seamlessly. For now I'm just
embedding them directly.

 working through this and learning along the way - will do my best to
document my "unboxing" learning experience and hopefully contribute in some
way -

I'll get back to SizeConstraint on my next commit so I can show you but I
think it might be ok with containersurfaces out of the mix

## 

Robert Moggach
Sr. VFX Supervisor / Flame Artist
Los Angeles / New York / London / Toronto
http://moggach.com
https://www.linkedin.com/profile/view?id=1664843
Mobile (310) 773-1793

**Comment by rjmoggach** [comment]
hey there - posted a famous-views question on stackoverflow - if you have a
moment would love some help...

http://stackoverflow.com/questions/28957965/famous-draggable-vs-genericsync-with-famous-views

also having some weird render issues when dragging to extents on my test4
view...

code is https://github.com/robmoggach/famous-views-testing/
site is http://mogfam.meteor.com

I'm happy to contribute from a noob perspective as I go through this
learning process.

## 

Robert Moggach
Sr. VFX Supervisor / Flame Artist
Los Angeles / New York / London / Toronto
http://moggach.com
https://www.linkedin.com/profile/view?id=1664843
Mobile (310) 773-1793
